### PR TITLE
fix timeout code in windows

### PR DESCRIPTION
**Pull request by jman-krafton**
Timeout was not working in Windows.

### CODE REVIEW

File: net.c [hunk 1]
```diff
@@ -63,7 +63,15 @@ int redisNetRead(redisContext *c, char *buf, size_t bufcap) {
         if ((errno == EWOULDBLOCK && !(c->flags & REDIS_BLOCK)) || (errno == EINTR)) {
             /* Try again later */
             return 0;
-        } else {
+        }
+		#ifdef _WIN32
+		else if(errno == ETIMEDOUT && (c->flags & REDIS_BLOCK))
+		{	/* especially in windows */
```
Comment (mnunberg) [inline net.c]: Can you fix the formatting here? (spaces, no tabs; opening brace on same line, etc.)

Comment (mnunberg) [reply net.c]: I also don't see how this block is specific to Windows.

File: net.c [hunk 2]
```diff
@@ -63,7 +63,15 @@ int redisNetRead(redisContext *c, char *buf, size_t bufcap) {
         if ((errno == EWOULDBLOCK && !(c->flags & REDIS_BLOCK)) || (errno == EINTR)) {
             /* Try again later */
             return 0;
-        } else {
+        }
+        #ifdef _WIN32
```
Comment (mnunberg) [inline net.c]: This doesn't need an `#ifdef`

Comment (mnunberg) [reply net.c]: also, fix brace on same line

File: net.c (hunk 1 above)
Comment (jman-krafton) [reply net.c]: > I also don't see how this block is specific to Windows.

In fact, I think so. I added the block in order to clarify a timeout error code when giving the timeout value on blocking mode. But, originally the block wasn't there and I was uncertain that I coded correct, so I specified the block to only Windows.

If this block doesn't have any problem, I can remove #preprocessor or else I will remove the block.

File: net.c (hunk 2 above)
Comment (jman-krafton) [reply net.c]: > also, fix brace on same line

i fixed it.

Comment (jman-krafton) [reply net.c]: > This doesn't need an `#ifdef`

i removed #ifdef ~ #endif

Comment (mnunberg) [reply net.c]: You're still using 8 space indentation.. are you using a proper editor? :)

**Comment by mnunberg** [comment]
Oh, and when you're done, please squash into two commits

File: net.c (hunk 2 above)
Comment (jman-krafton) [reply net.c]: My default indent setting(VS2017) is a tab. :]
I (really) have fixed all formatting mistakes.

**Comment by jman-krafton** [comment]
> Oh, and when you're done, please squash into two commits
I am done.

**Comment by HelloAWorld** [comment]
I aslo got timeout problem on windows, and I commit issues 
[redis timeout issues #677](https://github.com/redis/hiredis/issues/677)
Please review it if it's the same problem.

**Pull request merged**